pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/FilePerIndexDirectory.java

```diff
@@ -97,17 +98,14 @@ public void close()
 
   @Override
   public void removeIndex(String columnName, ColumnIndexType indexType) {
-    File indexFile = getFileFor(columnName, indexType);
-    if (indexFile.delete()) {
```

Comment:
1. file.delete() can't remove folder, while TextIndex is put in a folder. 
2. TextIndex has two parts: the index data in folder; and another mapping file.